app/src/main/java/com/kickstarter/viewmodels/ActivityFeedViewModel.java

```diff
@@ -1,38 +1,50 @@
 package com.kickstarter.viewmodels;
 
 import android.support.annotation.NonNull;
+import android.util.Pair;
 
 import com.kickstarter.libs.ActivityViewModel;
 import com.kickstarter.libs.ApiPaginator;
+import com.kickstarter.libs.CurrentConfigType;
 import com.kickstarter.libs.CurrentUserType;
 import com.kickstarter.libs.Environment;
-import com.kickstarter.libs.KoalaContext;
+import com.kickstarter.libs.FeatureKey;
+import com.kickstarter.libs.KoalaContext.Update;
 import com.kickstarter.libs.rx.transformers.Transformers;
 import com.kickstarter.libs.utils.ObjectUtils;
 import com.kickstarter.models.Activity;
 import com.kickstarter.models.Project;
+import com.kickstarter.models.SurveyResponse;
 import com.kickstarter.services.ApiClientType;
 import com.kickstarter.services.apiresponses.ActivityEnvelope;
 import com.kickstarter.ui.activities.ActivityFeedActivity;
-import com.kickstarter.ui.adapters.ActivityFeedAdapter;
+import com.kickstarter.ui.adapters.ActivityFeedAdapter.Delegate;
 import com.kickstarter.ui.viewholders.EmptyActivityFeedViewHolder;
 import com.kickstarter.ui.viewholders.FriendBackingViewHolder;
 import com.kickstarter.ui.viewholders.ProjectStateChangedPositiveViewHolder;
 import com.kickstarter.ui.viewholders.ProjectStateChangedViewHolder;
 import com.kickstarter.ui.viewholders.ProjectUpdateViewHolder;
+import com.kickstarter.ui.viewholders.UnansweredSurveyViewHolder;
+import com.trello.rxlifecycle.ActivityEvent;
 
+import java.util.Collections;
 import java.util.List;
 
 import rx.Observable;
 import rx.subjects.BehaviorSubject;
 import rx.subjects.PublishSubject;
 
+import static com.kickstarter.libs.utils.ObjectUtils.coalesce;
+
 public interface ActivityFeedViewModel {
 
-  interface Inputs extends ActivityFeedAdapter.Delegate {
+  interface Inputs extends Delegate {
     /** Invoke when pagination should happen. */
     void nextPage();
 
+    /** Invoke when activity's onResume runs */
+    void resume();
```

Comment:
so I suggested we create this input and hit it from the activity so that we could better test lifecycle methods, much like we do in ios.

this works, but one simplification we could do is to leverage the fact that all of our VM's have a base class, and they all have the lifecycle methods that activities go through, and so we could just make these lifecycle observables available for free! wanna try creating a `onResume` observable (let's call it `onResume` cause it won't conflict with the method `onResume()`) in the base class that is automatically pinged on resume, and then leverage that here? should mean this input goes away and the activity code goes away, but we get the same effect.